core/src/display_object/movie_clip.rs

```diff
@@ -1837,8 +1837,16 @@ impl<'gc> TDisplayObject<'gc> for MovieClip<'gc> {
         point: (Twips, Twips),
     ) -> bool {
         if self.world_bounds().contains(point) {
-            for child in self.iter_execution_list() {
-                if child.hit_test_shape(context, point) {
+            let mut clip_depth = 0;
+
+            for child in self.iter_render_list() {
```

Comment:
Seems like this needs to check for the the `masker` as well. (Sorry I didn't notice earlier!)

Reply:
Ah right, I forgot about that. I've added a test for visibility too.

Also, I noticed mask clips apparently respond to mouse events if they are in "button mode". I've added an argument to `hit_test_shape()` to handle that. Not sure if there's a better way to do this.